### PR TITLE
Issue #677 - Feature Request: Create build pipeline to build and publ…

### LINKED ISSUE #677: Feature Request: Create build pipeline to build and publish Exchange-API artifacts
Currently the Exchange-API container image is built and published by an IBM-internal Jenkins pipeline that make it difficult for open-source developers to see how its done. In addition, this is only done on a nightly basis with no way for non-IBM employees to trigger it.

We need to have a pipeline that builds and publishes the Exchange-API container image into Dockerhub. This pipeline should ideally take the form of Github Actions and operate off of pull requests. The pipeline should publish the testing tag to Dockerhub upon completion.

### PR DESCRIPTION
**Pull request by blakep7**
…ish Exchange-API artifacts

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Added GitHub action workflow 'build-push' to build docker image and push it to Dockerhub

**Necessary Repository Secrets and Variables**
_Found in: Repo Settings -> Secrets and Variables -> Actions_

- Secrets
   - DOCKER_USER: Username for docker login
   - DOCKER_TOKEN: Token for docker login
- Variables
   - DOCKERHUB_REPO: openhorizon

Fixes #677

### CODE REVIEW

File: .github/workflows/build-push.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Name of the workflow
+name: build-push
+
+# This workflow will run when there is a push (including merge) to the listed branches
+on:
+  push:
+    branches:
+      - master
+      - v2.87
```
Comment (bencourliss) [inline .github/workflows/build-push.yml:9]: we also need to add in branch `1.110` as well.

File: .github/workflows/build-push.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TKN }}
+
+      # Checkout our Github repo
+      - name: Checkout Github Repo
+        uses: actions/checkout@v3
+
+      # Setup Scala
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v13
```
Comment (bencourliss) [inline .github/workflows/build-push.yml:49]: who provides this action? We should document where it comes from and its source code repo.

**Review by bencourliss — CHANGES_REQUESTED** [review]

File: .github/workflows/build-push.yml (hunk 1 above)
Comment (bencourliss) [reply .github/workflows/build-push.yml:9]: also add it to the `run-tests.yml` pipeline as well.

**Review by bencourliss — CHANGES_REQUESTED** [review]

File: .github/workflows/build-push.yml (hunk 2 above)
Comment (naphelps) [reply .github/workflows/build-push.yml:49]: https://github.com/marketplace/actions/setup-scala
https://github.com/olafurpg/setup-scala

olafurpg/setup-scala has been deprecated as version 14. Recommendation from the author to move to coursier/setup-action.

https://github.com/marketplace/actions/coursier-setup-action
https://github.com/coursier/setup-action

File: .github/workflows/build-push.yml (hunk 1 above)
Comment (blakep7) [reply .github/workflows/build-push.yml:9]: I don't see a 1.110 branch. Is it named v1.110 or just 1.110?

File: .github/workflows/build-push.yml (hunk 2 above)
Comment (blakep7) [reply .github/workflows/build-push.yml]: Thanks @naphelps 

Updated to use [https://github.com/coursier/setup-action](https://github.com/coursier/setup-action), comment added to document source of action.

Comment (blakep7) [reply .github/workflows/build-push.yml]: Also changed the unit test workflow to use the new setup-action instead of the deprecated one.

**Review by naphelps — APPROVED** [review]
LGTM!

**Comment by blakep7** [comment]
Changed: Workflow uses secret 'DOCKER_TOKEN' now.

**Review by bencourliss — APPROVED** [review]

**Pull request merged**